src/averageAges.js

```diff
@@ -55,7 +69,19 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const children = onlyWithSon
+    ? people.filter(p => p.sex === 'm'
```

Comment:
If you filter the children array putting the respective child's mom, you avoid to looking for the mother again in the line 80 with find() statement. 
When you need the filter() and map() effects together, i.e. you need change the main array and skip some elements, is nice you use flatMap() function.
You can check the [reference docs](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/flatMap) if you want